lib/src/js/compiler.dart

```diff
@@ -0,0 +1,90 @@
+import 'dart:js_util';
+
+import 'package:node_interop/js.dart';
+
+import 'compile.dart';
+import 'compile_options.dart';
+import 'reflection.dart';
+import 'utils.dart';
+
+class Compiler {
+  bool _disposed = false;
+
+  void throwIfDisposed() {
+    if (_disposed) {
+      jsThrow(JsError('Compiler has already been disposed.'));
+    }
+  }
+}
+
+class AsyncCompiler extends Compiler {
+  final Set<Promise> _compilations = {};
```

Comment:
Since this member is accessed outside of its class (and the class is anyway), it should be public to help document that it's not exclusively class-internal.